From cb4e0750c836526d8ac918ba65bd758cf6c4b481 Mon Sep 17 00:00:00 2001
From: Michal Babej <michal.babej@tuni.fi>
Date: Fri, 27 Mar 2020 12:31:09 +0200
Subject: [PATCH] Fix some compiler warnings

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62514a179..6df8cee31 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,12 +60,12 @@ if(UNIX)
   if((CMAKE_C_COMPILER_ID STREQUAL "GNU") OR
      (CMAKE_C_COMPILER_ID STREQUAL "Clang"))
 
+    add_compile_options("$<$<COMPILE_LANGUAGE:C>:-Werror=implicit-function-declaration>")
+
     check_c_compiler_flag("-Wincompatible-pointer-types" HAVE_WARN_INCOMPATIBLE_POINTER_TYPES)
-    set(FORBIT_IMPLICIT_FUNCTIONS "-Werror=implicit-function-declaration")
     if (HAVE_WARN_INCOMPATIBLE_POINTER_TYPES)
-      set(FORBIT_IMPLICIT_FUNCTIONS ${FORBIT_IMPLICIT_FUNCTIONS} "-Werror=incompatible-pointer-types")
+      add_compile_options("$<$<COMPILE_LANGUAGE:C>:-Wincompatible-pointer-types>")
     endif()
-    add_compile_options(${FORBIT_IMPLICIT_FUNCTIONS})
 
   else()
     message(WARNING "Don't know how to forbid this compiler from allowing implicit function declarations.")
